### PR TITLE
Add api util to retrieve a transcript given a list of video ids.

### PR DESCRIPTION
**Pull request by Qubad786**

### CODE REVIEW

File: edxval/api.py [hunk 1]
```diff
@@ -190,6 +174,35 @@ def get_video_transcripts(video_id):
...
+        try:
+            video_transcript = VideoTranscript.objects.get(video_id=video_id, language_code=language_code)
+            transcript_content = video_transcript.transcript.file.read()
+            transcript_info = dict(file_name=video_transcript.file.name, content=transcript_content)
+            break
+        except VideoTranscript.DoesNotExist:
+            continue
+        except Exception:
+            logger.exception(
+                'Error while retrieving transcript for video=%s -- language_code=%s',
```
Comment (muzaffaryousaf) [inline edxval/api.py]: Log prefix would be helpful.

File: edxval/api.py [hunk 2]
```diff
@@ -158,22 +158,6 @@ def is_transcript_available(video_id, language_code=None):
     return transcript_set.exists()
 
 
-def get_video_transcript(video_id, language_code):
```
[the remark this answers is not in the document]
Comment (muzaffaryousaf) [reply edxval/api.py]: Probably, you just need to update it's usage i.e. sending a list of ids, the name is still same.

File: edxval/api.py [hunk 3]
```diff
@@ -190,6 +174,35 @@ def get_video_transcripts(video_id):
...
 
+def get_video_transcript(video_ids, language_code):
+    """
+    Get video transcript info
+
+    Arguments:
+        video_ids(list): list containing edx_video_id and external video ids extracted from
+        external sources from a video component.
+        language_code(unicode): it will the language code of the requested transcript.
+    """
```
[the remark this answers is not in the document]
Comment (muzaffaryousaf) [reply edxval/api.py]: will be the.

File: edxval/api.py [hunk 4]
```diff
@@ -190,6 +174,35 @@ def get_video_transcripts(video_id):
...
+    transcript_info = None
```
Comment (muzaffaryousaf) [inline edxval/api.py]: transcript_info --> transcript_data ?

File: edxval/tests/test_api.py [hunk 5]
```diff
@@ -1525,27 +1525,42 @@ def test_is_transcript_available(self, video_id, language_code, expected_availab
         is_transcript_available = api.is_transcript_available(video_id, language_code)
         self.assertEqual(is_transcript_available, expected_availability)
 
-    @unpack
+    @patch('edxval.api.logger')
+    def test_get_video_transcript_exception(self, mock_logger):
+        """
+        Verify that `get_video_transcript` logs and raises exceptioins.
```
Comment (muzaffaryousaf) [inline edxval/tests/test_api.py]: exceptioins --> exception.

File: edxval/models.py [hunk 6]
```diff
@@ -418,6 +418,22 @@ class Meta:
         unique_together = ('video_id', 'language_code')
 
     @classmethod
+    def get_or_none(cls, video_id, language_code):
+        """
+        Gets data model object and returns None if it does not exist.
```
Comment (muzaffaryousaf) [inline edxval/models.py]: Returns data model object if found or none otherwise.

File: edxval/models.py [hunk 7]
```diff
@@ -418,6 +418,22 @@ class Meta:
...
+
+        Arguments:
+            video_id(unicode): Video ID
+            language_code(unicode): Language Code
```
Comment (muzaffaryousaf) [inline edxval/models.py]: Well i prefer not to add args if we're just repeating their names.

File: edxval/api.py [hunk 8]
```diff
@@ -190,6 +174,35 @@ def get_video_transcripts(video_id):
     return transcripts
 
 
+def get_video_transcript(video_ids, language_code):
```
Comment (muzaffaryousaf) [inline edxval/api.py]: I think `get_video_transcript_content` or `get_video_transcript_data` makes more sense.

File: edxval/api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Qubad786) [reply edxval/api.py]: Its makes no sense for an api method to return mdoel object, so I;ve added a method which will bring you serialized transcript object.

[the remark this answers is not in the document]
Comment (Qubad786) [reply edxval/api.py]: you'll have to refactor the test.

**Comment by Qubad786** [comment]
@muzaffaryousaf, @muhammad-ammar  Its ready again

File: edxval/tests/test_api.py [hunk 9]
```diff
@@ -1585,7 +1603,10 @@ def test_create_video_transcript(self):
         )
         self.assertEqual(transcript_url, transcript_data['name'])
 
-        video_transcript = api.get_video_transcript(transcript_data['video_id'], transcript_data['language_code'])
+        video_transcript = VideoTranscript.objects.get(
+            video_id=transcript_data['video_id'],
+            language_code=transcript_data['language_code']
+        )
```
Comment (muzaffaryousaf) [inline edxval/tests/test_api.py]: You can  re-use the same method, since you've reverted the code.

File: edxval/tests/test_api.py [hunk 10]
```diff
@@ -1525,27 +1525,70 @@ def test_is_transcript_available(self, video_id, language_code, expected_availab
...
+    @data(
+        {'video_id': 'non-existant-video', 'language_code': 'en'},
+        {'video_id': '0987654321', 'language_code': 'en'},
+    )
     @unpack
+    def test_get_video_transcript_not_found(self, video_id, language_code):
+        """
+        Verify that `get_video_transcript_data` works as expected if transcript is not found.
+        """
+        self.assertIsNone(api.get_video_transcript(video_id, language_code))
```
Comment (muzaffaryousaf) [inline edxval/tests/test_api.py]: Where are you calling get_video_transcript_data ?

File: edxval/tests/test_api.py [hunk 11]
```diff
@@ -1525,27 +1525,70 @@ def test_is_transcript_available(self, video_id, language_code, expected_availab
...
+
+    def test_get_video_transcript(self):
+        """
+        Verify that `get_video_transcript_data` works as expected if transcript is found.
+        """
+        transcript = api.get_video_transcript(u'0987654321', u'ur')
```
Comment (muzaffaryousaf) [inline edxval/tests/test_api.py]: again not calling data method.

**Comment by Qubad786** [comment]
@muzaffaryousaf addressed.

File: edxval/api.py [hunk 12]
```diff
@@ -198,8 +230,7 @@ def get_video_transcript_url(video_id, language_code):
         video_id: it can be an edx_video_id or an external_id extracted from external sources in a video component.
         language_code: language code of a video transcript
     """
-    video_transcript = get_video_transcript(video_id, language_code)
-
+    video_transcript = VideoTranscript.get_or_none(video_id, language_code)
```
Comment (muzaffaryousaf) [inline edxval/api.py]: Don't we have url in serialized data ?

Comment (Qubad786) [reply edxval/api.py]: I think we should open a cleaning PR at the end to remove such unused api utils all together.

**Comment by Qubad786** [comment]
I'm merging this.

**Pull request merged**